packages/block-editor/src/store/private-selectors.js

```diff
@@ -18,3 +18,19 @@ export function isBlockInterfaceHidden( state ) {
 export function getLastInsertedBlocksClientIds( state ) {
 	return state?.lastBlockInserted?.clientIds;
 }
+
+export function getBlockEditingMode( state, clientId ) {
+	if ( ! clientId ) {
+		return state.settings.rootBlockEditingMode;
```

Comment:
I made this change in a recent commit—check it out. I kept it so that `block-editor` just exports a hook, though, instead of a `RootBlockEditingMode` component. That feels right to me as I don't want to be too prescriptive of how the hook is called.